### ISSUE TITLE
panic: send on closed channel

### ISSUE THREAD

**Issue by danielmai** [dgraph-io/dgraph#3687]
If you suspect this could be a bug, follow the template.

- What version of Dgraph are you using?

```
Dgraph version   : v1.0.12-rc3-603-g05eec2787
Commit SHA-1     : 05eec2787
Commit timestamp : 2019-07-17 15:24:51 -0700
Branch           : master
Go version       : go1.12.5
```

- Have you tried reproducing the issue with latest release?
Only tested with master.

- Steps to reproduce the issue (command/config used to run Dgraph).

1. Two-group cluster with 6 Alphas, 3 Zeros, `--replicas=3`.

```
./compose -a 6 -z 3
```

2. Run Flock.

```
./flock -a ":9180,:9182,:9183,:9184,:9185,:9186"
```

- Actual result

Panic happened on two of the followers from group 2.

```
E0718 03:30:13.061641       1 writer.go:49] TxnWriter got error during callback: Writes are blocked, possibly due to DropAll or Close
I0718 03:30:13.061655       1 log.go:34] Rolling up Created batch of size: 945 kB in 17.137071ms.
I0718 03:30:13.317643       1 log.go:34] Rolling up Time elapsed: 02s, bytes sent: 9.7 MB, speed: 4.9 MB/sec
I0718 03:30:13.317656       1 log.go:34] Rolling up Sent 85892 keys
E0718 03:30:13.061666       1 writer.go:49] TxnWriter got error during callback: Writes are blocked, possibly due to DropAll or Close
E0718 03:30:13.327026       1 draft.go:408] Error while rolling up lists at 270274: Writes are blocked, possibly due to DropAll or Close
I0718 03:30:16.000581       1 log.go:34] LOG Compact 0->1, del 3 tables, add 2 tables, took 2.956439943s
I0718 03:30:16.000627       1 log.go:34] Compaction for level: 0 DONE
I0718 03:30:17.847709       1 log.go:34] LOG Compact 1->1, del 1 tables, add 1 tables, took 1.847039765s
I0718 03:30:17.847742       1 log.go:34] DropPrefix done
I0718 03:30:17.847759       1 log.go:34] Resuming writes
I0718 03:30:17.847797       1 schema.go:78] Deleting schema for predicate: [user_id]
I0718 03:31:11.288867       1 draft.go:1201] Num pending txns: 7
I0718 03:31:11.307005       1 draft.go:664] [0x5] Set Raft progress to index: 228323.
I0718 03:32:11.288734       1 draft.go:1201] Num pending txns: 6
I0718 03:32:11.328315       1 draft.go:664] [0x5] Set Raft progress to index: 230607.
I0718 03:33:10.570176       1 draft.go:370] Creating snapshot at index: 232440. ReadTs: 285638.
I0718 03:33:10.570204       1 storage.go:589] CreateSnapshot i=232440, cs=nodes:4 nodes:5 nodes:6 
I0718 03:33:11.288852       1 draft.go:1201] Num pending txns: 3
I0718 03:33:11.293827       1 draft.go:664] [0x5] Set Raft progress to index: 232463.
W0718 03:33:33.027598       1 draft.go:950] Raft.Ready took too long to process: Timer Total: 188ms. Breakdown: [{sync 188ms} {disk 0s} {proposals 0s} {advance 0s}] Num entries: 3. MustSync: true
I0718 03:34:11.288725       1 draft.go:1201] Num pending txns: 5
I0718 03:34:11.302953       1 draft.go:664] [0x5] Set Raft progress to index: 234095.
I0718 03:35:11.288788       1 draft.go:1201] Num pending txns: 2
I0718 03:35:11.303422       1 draft.go:664] [0x5] Set Raft progress to index: 235587.
I0718 03:35:11.849765       1 log.go:34] Rolling up Created batch of size: 4.2 MB in 303.407218ms.
I0718 03:35:12.276168       1 log.go:34] Rolling up Created batch of size: 3.5 MB in 255.018688ms.
I0718 03:35:12.288803       1 log.go:34] Rolling up Time elapsed: 01s, bytes sent: 7.8 MB, speed: 7.8 MB/sec
I0718 03:35:12.573027       1 index.go:969] Dropping predicate: [user_id]
I0718 03:35:12.573057       1 log.go:34] DropPrefix called on 00000000  00 00 07 75 73 65 72 5f  69 64                    |...user_id|
. Blocking writes...
panic: send on closed channel

goroutine 6085484 [running]:
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*DB).sendToWriteCh(0xc0000cea80, 0xc00c7d73d0, 0x2, 0x2, 0x6, 0x8, 0x13)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/db.go:681 +0x177
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Txn).commitAndSend(0xc00ca8a400, 0x0, 0x0, 0x0)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/txn.go:495 +0x461
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Txn).CommitWith(0xc00ca8a400, 0xc00c7d73c0)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/txn.go:599 +0x12b
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Txn).CommitAt(0xc00ca8a400, 0x3f87b, 0xc00c7d73c0, 0x0, 0x0)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/managed_db.go:56 +0x54
github.com/dgraph-io/dgraph/posting.(*TxnWriter).update(0xc00983d3e0, 0x3f87b, 0xc0093f5ab0, 0x0, 0x0)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/posting/writer.go:84 +0x12f
github.com/dgraph-io/dgraph/posting.(*TxnWriter).SetAt(0xc00983d3e0, 0xc011b0e780, 0x1f, 0x20, 0xc011b0e7a0, 0x14, 0x14, 0x8, 0x3f87b, 0x0, ...)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/posting/writer.go:89 +0xc3
github.com/dgraph-io/dgraph/posting.(*TxnWriter).Write(0xc00983d3e0, 0xc023182700, 0x11abbfa9, 0x2e7da407a64ca)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/posting/writer.go:63 +0xc9
github.com/dgraph-io/dgraph/worker.(*node).rollupLists.func4(0xc023182700, 0x647793feb29, 0x21fc860)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/worker/draft.go:1022 +0x34
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Stream).streamKVs.func1(0xc023182700, 0xc0093f5d8c, 0x3)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/stream.go:267 +0x299
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Stream).streamKVs(0xc02ab88850, 0x18bf8c0, 0xc0000fc020, 0x0, 0x0)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/stream.go:298 +0x65a
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Stream).Orchestrate.func2(0xc02959cd80, 0xc02ab88850, 0x18bf8c0, 0xc0000fc020)
        /home/dmai/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/stream.go:349 +0x3f
created by github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*Stream).Orchestrate
        /home/dmai/go/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/stream.go:347 +0x1ed
```

**Comment by jarifibrahim** [comment]
This has been fixed in badger https://github.com/dgraph-io/badger/pull/918.

**Comment by campoy** [comment]
Was this included in v1.0.16, @danielmai?
Will the fix be included in v1.1?

**Comment by danielmai** [comment]
We need to bring this into v1.1, which will include the Badger data format changes.

I haven't seen if this happens in v1.0.16. Both master and v1.0.16 are currently using Badger v1.6.0 (master: https://github.com/dgraph-io/badger/commit/50ccc861a93d4f68fb4c0698ee6c4ffeb300229a, v1.0.16: https://github.com/dgraph-io/badger/commit/2fa005c9d4bf695277ab5214c1fbce3735b9562a, the documentation commit afterwards), so v1.0.16 could see this issue too.

**Comment by mangalaman93** [comment]
Does this issue still exist?

**Comment by danielmai** [comment]
In Dgraph master, Badger master branch as of Aug 2nd (f59246cb119d2f7b6f2fc29a835ee5edb366019a) has been vendored, so the bug fix is already in Dgraph v1.1.
